### DISCUSSION TITLE
"Offline conversions" like GA measurement protocol?

### DISCUSSION THREAD

**Discussion by xporterfield** [plausible/analytics#841] [Anything Else]
Hello,

I am looking for a Google Analytics alternative, and Plausible looks like a great fit -- although I would need the ability to track conversions or goals from users after they convert on a third party.  Postbacks or some sort of pixel is how I usually do it with the Google Analytics Measurement Protocol, but I am wondering if it's possible have the same functionality with Plausible?

**Comment by metmarkosaric** [comment]
thanks @xporterfield! can you explain a bit more in detail what you are trying to do?

**Comment by ukutaht** [comment]
I don't think there's any fundamental limitation but we don't have an HTTP API at the moment (It's coming).

The main question for me is how we can correlate the user ID. How does it work with GA, how do you acquire the user id to send along with the conversion event?

**Reply by xporterfield** [reply]
For Google Analytics, I am using their measurement protocol to send all pageviews with custom generated clientIDs, the cid parameter.  That is, an md5 hash of (user_agent + do_not_track setting + user IP) for each pageview hit.  (I switched to this to avoid Google's cookie version _ga client ID).

Here is a good reference, it's a similar setup (but I am using nginx instead of php) https://wordpress.org/plugins/ga-backend-tracking/

My outbound affiliate links have this clientID passed as a parameter, and on conversion, a postback to my server is triggered.   This postback is then forwarded to the GA measurement protocol with the cid and conversion value.

**Reply by ukutaht** [reply]
OK basically from our end it seems we just need to accept a `user_id`  parameter that overrides our ID generation on the backend.
